core/src/main/java/org/apache/carbondata/core/scan/expression/Expression.java

```diff
@@ -33,6 +33,8 @@
   protected List<Expression> children =
       new ArrayList<Expression>(CarbonCommonConstants.DEFAULT_COLLECTION_SIZE);
 
+  protected boolean isAlreadyResolved;
```

Comment:
Add doc for this variable when isAlreadyResolved will be true

Reply:
Done.